### ISSUE TITLE
[Linux]: Window is transparent when viewing image with transparent background

### ISSUE THREAD

**Issue by yuqiang-yuan** [zed-industries/zed#18800]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When open a image with transparent background, e.g. `ico` or `png`, the editor window is transparent.
![Screenshot from 2024-10-07 16-47-15](https://github.com/user-attachments/assets/f4420442-a277-4d01-9d68-ddf94adb37c3)

### Environment

Zed: v0.155.2 (Zed)
OS: Linux X11 ubuntu 24.04
Memory: 31.2 GiB
Architecture: x86_64
GPU: NVIDIA GeForce RTX 4060 Ti || NVIDIA || 535.183.01

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by jansol** [comment]
Can confirm on Wayland as well, the checkerboard background of the image viewer is semi-transparent regardless of whether or not the theme has a transparent background.

**Comment by bbb651** [comment]
The [checkerboard drawing](https://github.com/zed-industries/zed/blob/f50bca76303f958870f4a3e8864647a63194ebce/crates/image_viewer/src/image_viewer.rs#L189-L211) is also [affecting alpha](https://github.com/zed-industries/zed/blob/f50bca76303f958870f4a3e8864647a63194ebce/crates/gpui/src/color.rs#L322-L330), I don't see a good reason why it should do that on any platform, so I think changing `a` to `1.` should fix it?

**Comment by jansol** [comment]
The bigger question here IMHO is why it overrides/skips the alpha of the window background, which should be 1.0 in non-transparent themes. Blending a semi-transparent checkerboard on top of the window background does still make some sense (it tints the checkerboard), although I'd probably want the checkerboard to be configurable in terms of opacity and brightness/contrast either way.

**Comment by bbb651** [comment]
I don't think it overrides the alpha of the window? I cannot reproduce it when setting `background` to `#xxxxxxff`. Unlike other platforms, on wayland `background.appearance` is only used as an hint for the compositor if the background is transparent and doesn't disable transparency, which is probably the real bug (it can also cause broken visuals when it's set to `opaque` and transparency is less than `1.0` because the compositor is allowed to assume that's impossible).
Also, it's a little odd that `background` blends there's, but not on e.g. the status bar..
![image](https://github.com/user-attachments/assets/0128697a-15d3-41e3-b040-d808565544ac)

**Comment by jansol** [comment]
Hm, that's weird. That would imply that the builtin Ayu Dark theme I checked with does not specify an opaque background color - but it does.

**Comment by yuqiang-yuan** [comment]
No this issue on Zed 0.177. issue closed
